src/index.d.ts

```diff
@@ -491,13 +491,12 @@ declare namespace admin.auth {
     /**
      * The date the user last signed in, formatted as a UTC string.
      */
-    lastSignInTime: string;
+    lastSignInTime: string | null;
```

Comment:
Nit: Let's also update the doc comment to indicate when these values are null.

Reply:
Done. Though note that there's not a great explanation as to why creationTime might be null. As an alternative, we could keep it as non-nullable and change the implementation to throw an internal error if this situation occurs.

Reply:
That sounds like the right approach to me.

Reply:
Done. A number of unit tests needed to be changed as a result. I think the changes are correct, but definitely will want Bassam to weigh in on this too.